cmd/kubeadm/app/features/features.go

```diff
@@ -22,22 +22,66 @@ import (
 	"strconv"
 	"strings"
 
-	utilfeature "k8s.io/apiserver/pkg/util/feature"
```

Comment:
why remove this dep?

Reply:
yeah, I asked that myself. But we're not actually using that much from it, just a few string types, so it doesn't add much value TBH. If we embedded those structs, we'd need to start doing stuff like 

```
HighAvailability:    {utilfeature.FeatureSpec{Default: false, PreRelease: alpha}, MinimumVersion: v190},
```

which looks a bit weird imo.

Reply:
I don't really mind either, we can keep on using it if somebody has a strong preference

Reply:
I would like to stay as sync as possible with upstream, please.

Reply:
Sure, I will add back those types. However I don't think there is consistent tooling around this for "upstream". Here we're just cribbing off what apiserver does.

Reply:
heh, can be generally improved indeed, yes